prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -103,194 +102,24 @@ def smart_charging(
-            load_demand[i] + model_year_profile[i]
-            for i in range(
-                day_iter * 24, (day_iter + 1) * 24 + min(day_num - day_iter - 1, 2) * 24
-            )
-        ]
-
-        if 3 - day_num + day_iter > 0:
-            adjusted_load += [
-                load_demand[i] + model_year_profile[i]
-                for i in range(24 * (3 - day_num + day_iter))
-            ]
-
-        cost = np.array(adjusted_load)
-
-        g2v_load = np.zeros((100, 72))
-        individual_g2v_load = np.zeros((nd_len, 72))
-
-        i = 0
-
-        while i < nd_len:
-
-            # trip amount for each vehicle
-            total_trips = int(newdata.iloc[i, newdata.columns.get_loc("total_trips")])
-
-            if data_day[i] == input_day[day_iter]:
-
-                # only home based trips
-                if (
-                    newdata.iloc[i, newdata.columns.get_loc("why_from")]
-                    * newdata.iloc[
-                        i + total_trips - 1, newdata.columns.get_loc("dwell_location")
-                    ]
-                    == 1
-                ):
-
-                    # copy one vehicle information to the block
-                    individual = newdata.iloc[i : i + total_trips].copy()
-
-                    individual["rates"] = individual.apply(
-                        lambda d: dwelling.get_rates(
-                            cost,
-                            d["trip_end"],
-                            d["dwell_time"],
-                        ),
-                        axis=1,
-                    )
-
-                    charging_consumption = individual["charging consumption"].to_numpy()
-
-                    rates = individual["rates"]
-                    rates = [r for trip_rates in rates for r in trip_rates]
-
-                    elimit = individual["energy limit"]
-                    elimit = [el for energy_lim in elimit for el in energy_lim]
-
-                    seg = individual["seg"].apply(int).to_numpy()
-
-                    linprog_inputs = charging_optimization.calculate_optimization(
-                        charging_consumption,
-                        rates,
-                        elimit,
-                        seg,
-                        total_trips,
-                        kwh,
-                        charging_efficiency,
-                    )
-
-                    linprog_result = linprog(**linprog_inputs)
-
-                    # fval is the value of the final cost, exitflag is the reason why the optimization terminates
-                    # 0-success, 1-limit reached, 2-problem infeasible, 3-problem unbounded, 4-numerical difficulties
-                    x = np.array(linprog_result.x)
-                    exitflag = linprog_result.status
-
-                    state_of_charge = np.zeros((total_trips, 2))
-
-                    # find the feasible points
-                    if exitflag == 0:
+        adjusted_load = load_demand + model_year_profile
```

Comment:
Wonder what happened here, did we preprocessed `load_demand` and `model_year_profile` somewhere else? Also, if that's the case should we move it out of the loop given it is not affected by the iterator `day_iter`.

Reply:
It's affected by the iteration, this line is adding current state of the model year profile to the base load. Here's the new simplified daily loop:

<img width="731" alt="Screen Shot 2022-11-29 at 12 32 13 PM" src="https://user-images.githubusercontent.com/1626883/204641871-a85a127d-5eed-4cf7-bcf0-f92b5931f6eb.png">

Reply:
I still don't see how `adjusted_load` is changed in the iteration. It seems we are only slicing it with different `trip_window_indices`. Also, in `calculate_daily_smart_charging_trips`, it seems only the size of `adjusted_load` matters?

Reply:
I see. I was wondering how the refactored logic is equivalent to the original implementation.